### PR TITLE
Refactor MQTT

### PR DESCRIPTION
**Pull request by shinsuke-nara**
Refactoring fixes in #102 is separated to this PR.

In this PR, I fixed 2 points.

  - If receiving push notification is failed, retry to connect MQTT
    server with already retrived endpoint information.
  - Refactor kiiPush_recvMsgTask.

In review of #102, We need to fix following 2 points.

  - kiiPush_retrievingEndpointResult_e -> kiiPush_retrieveEndpointResult
  - kiiPush_receivingState_e -> kiiPush_state

These will be fixed in this PR.

Related issue #102

### CODE REVIEW

**Comment by shinsuke-nara** [comment]
- kiiPush_retrievingEndpointResult_e -> kiiPush_retrieveEndpointResult
 - kiiPush_receivingState_e -> kiiPush_state

These were fixed.

File: kii/kii_push.c [hunk 1]
```diff
@@ -13,10 +13,17 @@
 
 typedef enum
 {
-    KIIPUSH_ENDPOINT_READY = 0,
-    KIIPUSH_ENDPOINT_UNAVAILABLE = 1,
-    KIIPUSH_ENDPOINT_ERROR = 2
-} kiiPush_endpointState_e;
+    KIIPUSH_RETRIEVING_ENDPOINT_SUCCESS = 0,
```
Comment (satoshikumano) [inline kii/kii_push.c]: RETRIEVE

File: kii/kii_push.c [hunk 2]
```diff
@@ -13,10 +13,17 @@
...
+    KIIPUSH_RETRIEVING_ENDPOINT_RETRY = 1,
+    KIIPUSH_RETRIEVING_ERROR = 2
```
Comment (satoshikumano) [inline kii/kii_push.c]: KIIPUSH_RETRIEVE_ENDPOINT_ERROR

File: kii/kii_push.c [hunk 3]
```diff
@@ -323,161 +386,155 @@ static void* kiiPush_recvMsgTask(void* sdata)
...
-                    {
-                        M_KII_LOG(kii->kii_core.logger_cb("kii-error: mqtt buffer overflow\r\n"));
-                        kii->_mqtt_endpoint_ready = 0;
-                        continue;
-                    }
+static void* kiiPush_recvMsgTask(void* sdata)
+{
+    kii_t* kii;
+    kii_mqtt_endpoint_t endpoint;
+    kiiPush_state receivingState = KIIPUSH_PREPARING_ENDPOINT;
```
Comment (satoshikumano) [inline kii/kii_push.c]: pushState

File: kii/kii_push.c [hunk 4]
```diff
@@ -323,161 +386,155 @@ static void* kiiPush_recvMsgTask(void* sdata)
...
-                            M_KII_LOG(kii->kii_core.logger_cb("Successfully Recieved Push %s\n", p));
-                            callback(kii, p, remainingLen - 2 - topicLen);
-                        }
-                        else
-                        {
-                            M_KII_LOG(kii->kii_core.logger_cb("kii-error: mqtt topic length error\r\n"));
-                            kii->_mqtt_endpoint_ready = 0;
-                            continue;
-                        }
+                        kii->_mqtt_endpoint_ready = 1;
```
Comment (shinsuke-nara) [inline kii/kii_push.c]: _mqtt_endpoint_ready -> _mqtt_connected

File: kii/kii_push.c (hunk 1 above)
Comment (shinsuke-nara) [inline kii/kii_push.c]: I fixed this.

File: kii/kii_push.c (hunk 2 above)
Comment (shinsuke-nara) [inline kii/kii_push.c]: I fixed this.

File: kii/kii_push.c (hunk 3 above)
Comment (shinsuke-nara) [inline kii/kii_push.c]: I fixed this.

**Comment by shinsuke-nara** [comment]
@satoshikumano I fixed this PR according to your comment. Please review this.

File: kii/kii_push.c [hunk 5]
```diff
@@ -13,10 +13,17 @@
 
 typedef enum
 {
-    KIIPUSH_ENDPOINT_READY = 0,
-    KIIPUSH_ENDPOINT_UNAVAILABLE = 1,
-    KIIPUSH_ENDPOINT_ERROR = 2
-} kiiPush_endpointState_e;
+    KIIPUSH_RETRIEVE_ENDPOINT_SUCCESS = 0,
+    KIIPUSH_RETRIEVE_ENDPOINT_RETRY = 1,
+    KIIPUSH_RETRIEVE_ERROR = 2
```
Comment (satoshikumano) [inline kii/kii_push.c]: KIIPUSH_RETRIEVE_ENDPOINT_ERROR

Comment (shinsuke-nara) [inline kii/kii_push.c]: I fixed this with f0b847f.

**Comment by shinsuke-nara** [comment]
@satoshikumano I fixed this PR. Could you review this?

**Pull request merged**